### ISSUE TITLE
Adding basic auth

### ISSUE THREAD

**Issue by pdincau** [beam-italia/elixir-plug-api-example#5]
I would like to add basic auth to this example. What do you think? @gpad @christian-fei

**Comment by christian-fei** [comment]
Yes please, it would be a great example for testing a plug

**Comment by christian-fei** [comment]
On a sidenote:
we could rename the repo to something more generic:

elixir-api-example

or more specific

elixir-plug-ecto-example

Input?

**Comment by pdincau** [comment]
@christian-fei could you open a separate issue for that?

**Comment by christian-fei** [comment]
:+1:

**Comment by pdincau** [comment]
I will work on this one.

**Comment by christian-fei** [comment]
nice!

**Comment by pdincau** [comment]
added the feature in a separate branch. You can see it here: eb8da204c2dd9d71b3f445bbb717b36c569c80f0

Comments?

**Comment by christian-fei** [comment]
@pdincau 
The only thing that's a bit strange are the tests on the router.
Somehow it seems you're testing the plug itself, or just to check that everything fits together as expected?

**Comment by pdincau** [comment]
@christian-fei true. But if the plug is added then you have to put the correct headers to go on with the request. I am wondering whether the user and password can be extracted from the config file.

**Comment by pdincau** [comment]
seems that with this plug you can't read from configuration. I will double check it

**Comment by christian-fei** [comment]
I think you can only configure applications http://elixir-lang.org/getting-started/mix-otp/distributed-tasks-and-configuration.html#application-environment-and-configuration

**Comment by christian-fei** [comment]
In this case, as the user of the lib, you will need to provide one yourself via `Application.get_env/3` like

```
Application.get_env(:api, :basic_auth)
```

The third param is the default value if the key is not found

and in the configuration file

```
config :api, :basic_auth,
  username:      "s3cr3t",
  password:      "s3cr3t",
```

**Comment by pdincau** [comment]
Implemented in 864e5889521c8fae541f94247d5df6f03a2fdb61
